chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/InvalidCriteriaCombinationReviewer.java

```diff
@@ -1,68 +1,66 @@
 package gov.healthit.chpl.validation.listing.reviewer.edition2015;
 
+import java.util.Iterator;
 import java.util.Optional;
 
 import org.ff4j.FF4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 import gov.healthit.chpl.FeatureList;
+import gov.healthit.chpl.domain.CertificationCriterion;
 import gov.healthit.chpl.domain.CertificationResult;
 import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
 import gov.healthit.chpl.util.ErrorMessageUtil;
+import gov.healthit.chpl.validation.InvalidCriteriaCombination;
 import gov.healthit.chpl.validation.listing.reviewer.Reviewer;
 
 @Component("invalidCriteriaCombinationReviewer")
-public class InvalidCriteriaCombinationReviewer implements Reviewer {
-
-    @Value("${criterion.170_315_b_6}")
-    private Integer criteriaB6Id;
-
-    @Value("${criterion.170_315_b_10}")
-    private Integer criteriaB10Id;
-
-    @Value("${criterion.170_315_g_8}")
-    private Integer criteriaG8Id;
-
-    @Value("${criterion.170_315_g_10}")
-    private Integer criteriaG10Id;
-
-    private ErrorMessageUtil msgUtil;
-    private FF4j ff4j;
-
+public class InvalidCriteriaCombinationReviewer extends InvalidCriteriaCombination implements Reviewer {
     @Autowired
     public InvalidCriteriaCombinationReviewer(ErrorMessageUtil msgUtil, FF4j ff4j) {
-        this.msgUtil = msgUtil;
-        this.ff4j = ff4j;
+        super(msgUtil, ff4j);
     }
 
     @Override
     public void review(CertifiedProductSearchDetails listing) {
         if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
             checkForInvalidCriteriaCombination(listing, criteriaB6Id, criteriaB10Id);
             checkForInvalidCriteriaCombination(listing, criteriaG8Id, criteriaG10Id);
+
+            initializeOldAndNewCriteria();
+            Iterator<Integer> oldCriteriaIdsIter = oldCriteriaIds.iterator();
+            Iterator<Integer> newCriteriaIdsIter = newCriteriaIds.iterator();
+            while (oldCriteriaIdsIter.hasNext() && newCriteriaIdsIter.hasNext()) {
```

Comment:
I'm uncomfortable with this setup. This assumes each array has the correct order of paired criteria to check against, and while it looks like the "initialize" method sets that up, it's hard to parse that. What if, instead, you created an array of Tuples/Pairs, maybe using something like https://docs.spring.io/spring-data/data-commons/docs/current/api/org/springframework/data/util/Pair.html, which would actually bind the old & new criteria together in a single object, and then allow those to be iterated over and help together?

Reply:
I agree and was going to comment something similar but your suggestion is a good one and leaves most of the code how it's been written. I was going to suggest not using the IDs at all and just coding a more generic check that says the listing can't attest to 2 criterion of the same number.

Reply:
Two of the same number would work, but that wouldn't let us roll the existing "can't have b6 and b10" or "g8 and g10" into the same philosophy